exchanges/bybit/bybit_websocket.go

```diff
@@ -0,0 +1,483 @@
+package bybit
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/common/crypto"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/orderbook"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/stream"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/ticker"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/trade"
+	"github.com/thrasher-corp/gocryptotrader/log"
+)
+
+const (
+	bybitWSBaseURL      = "wss://stream.bybit.com/"
+	wsSpotPublicTopicV2 = "spot/quote/ws/v2"
+	wsSpotPrivate       = "spot/ws"
+	bybitWebsocketTimer = 30 * time.Second
+	wsOrderbook         = "depth"
+	wsTicker            = "bookTicker"
+	wsTrades            = "trade"
+	wsKlines            = "kline"
+
+	wsAccountInfoStr = "outboundAccountInfo"
+	wsOrderStr       = "executionReport"
+	wsOrderFilledStr = "ticketInfo"
+
+	sub    = "sub"    // event for subscribe
+	cancel = "cancel" // event for unsubscribe
+)
+
+var comms = make(chan stream.Response)
+
+// WsConnect connects to a websocket feed
+func (by *Bybit) WsConnect() error {
+	if !by.Websocket.IsEnabled() || !by.IsEnabled() {
```

Comment:
Could you please also add a check for the SPOT asset being enabled?
I have SPOT disabled in my config, however I get:
```
[INFO]  | WEBSOCKET |  05/05/2022 09:58:42  | Bybit Websocket connected to wss://stream.bybit.com/spot/quote/ws/v2
[DEBUG] | WEBSOCKET |  05/05/2022 09:58:42  | Bybit websocket: running connection monitor cycle
```
Which we don't want unless we have it enabled

Reply:
Thanks for that. I won't actually request any further changes with this. It still runs when Spot is disabled, but changing this is not a feature we do elsewhere